emails/views.py

```diff
@@ -598,12 +598,20 @@ def _reply_allowed(from_address, to_address, reply_record):
 
 def _handle_reply(from_address, message_json, to_address):
     mail = message_json["mail"]
-    (lookup_key, encryption_key) = _get_keys_from_headers(mail["headers"])
-    reply_record = _get_reply_record_from_lookup_key(lookup_key)
+    try:
+        (lookup_key, encryption_key) = _get_keys_from_headers(mail["headers"])
+    except InReplyToNotFound:
+        return HttpResponse("No In-Reply-To header", status=400)
+
+    try:
+        reply_record = _get_reply_record_from_lookup_key(lookup_key)
+    except Reply.DoesNotExist:
+        return HttpResponse("Unknown or stale In-Reply-To header", status=404)
```

Comment:
Nit: adding a metric (via Markus, logger, Sentry, etc.) would be helpful to later make decision on if this is be abused/attacked.

Reply:
I'm adding a metric for now. I think this code will need to be refactored for the task backend (no more HTTP responses), and that will be a good time to tackle logs.